### PR TITLE
Support allow flip mode

### PR DESCRIPTION
**Pull request by cquiroz**
This PR adds support for the `Allow flip` mode in AGS. Besides the algorithm changes there are some changes on the UI which are relevant:
* The pos angle control is now disabled while calculations are taking place
* Eventually if the flip mode needs to change the PA the value will be directly updated on the controls

The new feature can be seen here:

https://user-images.githubusercontent.com/3615303/208494418-b13e3c14-60a5-4862-a835-6a7380ec3d34.mp4

### CODE REVIEW

**Review by rpiaggio — APPROVED** [review]
LGTM

File: explore/src/main/scala/explore/targeteditor/AladinCell.scala [hunk 1]
```diff
@@ -179,41 +195,39 @@ object AladinCell extends ModelOptics with AladinCommon:
...
       // to get faster reusability use a serial state, rather than check every candidate
       .useSerialState(List.empty[GuideStarCandidate])
       // Analysis results
       .useSerialState(List.empty[AgsAnalysis])
-      // Ags state
-      .useState[AgsState](AgsState.Idle)
       // Request data again if vizTime changes more than a month
-      .useEffectWithDepsBy((p, _, _, _, _, _, _) => p.obsConf.vizTime) {
-        (props, ctx, _, _, gs, _, agsState) => vizTime =>
+      .useEffectWithDepsBy((p, _, __, _, _) => p.obsConf.vizTime) {
```
Comment (toddburnside) [inline explore/src/main/scala/explore/targeteditor/AladinCell.scala:205]: Is this two `_` side by side? I didn't know that would work.

**Review by toddburnside — APPROVED** [review]
LGTM.

File: explore/src/main/scala/explore/targeteditor/AladinCell.scala (hunk 1 above)
Comment (cquiroz) [reply explore/src/main/scala/explore/targeteditor/AladinCell.scala:205]: oh weird, it certainly compiles

**Pull request merged**